src/code-generator/entry-points/non-interactive-cli.ts

```diff
@@ -1,6 +1,7 @@
 import { AppError } from "../error-handling";
 import { factorDefaultOptions } from "../generation-logic/generation-options";
 import { generateApp } from "../generation-logic/generate-service";
+import { spinner, cliTexts } from "../utils";
```

Comment:
Two notes here:

1. Our CLI has two main 'layers': entry-point (UI) and logic. These 'utils' are not generic rather ui-elements, hence they should get placed within the entry-points folder
2. I would avoid names like 'utils'. Think about the reader of the code, 'utils' kinda tell her - there are multiple things here which you'll have to find out yourself. Consider a name that actually tells what's in there like 'ui-elements'. You may also break the entry-point to two sub-folders: interactiveCLI and nonInteractiveCLI (what we handle now), then put all of those files in the nonInteractiveCLI

Reply:
Nice point, I agree with you. I am pushing a new folder structure.